kinto/core/resource/__init__.py

```diff
@@ -717,7 +717,8 @@ def _raise_400_if_invalid_id(self, record_id):
 
         :raises: :class:`pyramid.httpexceptions.HTTPBadRequest`
         """
-        if not self.model.id_generator.match(six.text_type(record_id)):
+        is_string = isinstance(record_id, six.text_type)
```

Comment:
The ID in the URL will always been seen as a string? If this is just to fix the bug in the body then I misunderstood the scope of this PR and you can discard this comment.